sysbuild/mcuboot.cmake

```diff
@@ -4,25 +4,49 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 
 if(SB_CONFIG_MCUBOOT_BUILD_DIRECT_XIP_VARIANT)
-  set(image mcuboot_secondary_app)
-  if(SB_CONFIG_PARTITION_MANAGER)
-    ExternalNcsVariantProject_Add(APPLICATION ${DEFAULT_IMAGE} VARIANT ${image})
+  get_property(images GLOBAL PROPERTY sysbuild_images)
 
-    set_property(GLOBAL APPEND PROPERTY
-        PM_APP_IMAGES
-        "${image}"
-    )
-  else()
-    # TODO: NCSDK-33774 add a general way to pass configuration options to the variant
-    # image
-    zephyr_get(extra_conf_file SYSBUILD LOCAL VAR EXTRA_CONF_FILE)
+  foreach(primary_image ${images})
+    # Do not build variant projects for MCUboot as well as MCUboot secondary slot.
+    if(${primary_image} STREQUAL "mcuboot")
+      continue()
+    elseif(${primary_image} STREQUAL "s1_image")
+      continue()
+    elseif(${primary_image} STREQUAL ${DEFAULT_IMAGE})
+      # Keep the old name for backward compatibility.
+      set(image "mcuboot_secondary_app")
+    elseif(SB_CONFIG_SOC_SERIES_NRF53X)
+      # Direct XIP on the radio image is not supported on nRF5340 SoC.
+      continue()
+    else()
```

Comment:
as above, no on this, it also completely breaks if someone changes a name of an application